### PR TITLE
[transaction-block-rename][2/n] rename transaction data structures

### PR DESCRIPTION
**Pull request by 666lcz**
## Description 

SuiTransaction* -> SuiTransactionBlock*

- SuiTransactionResponse → SuiTransactionBlockResponse
- SuiTransactionResponseQuery → SuiTransactionBlockResponseQuery
- SuiTransactionResponseOptions → SuiTransactionBlockResponseOptions
- SuiTransaction → SuiTransactionBlock
- SuiTransactionData → SuiTransactionBlockData
- SuiTransactionKind → SuiTransactionBlockKind
- SuiProgrammableTransaction → SuiProgrammableTransactionBlock

TransactionBytes → TransactionBlockBytes
TransactionsPage → TransactionBlocksPage

## Next PRs
- SuiCommand → SuiTransaction
- update Serde Rename
- rename RPC endpoint names (and TS usage)
## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/typescript/src/signers/signer-with-provider.ts [hunk 1]
```diff
@@ -82,29 +82,29 @@ export abstract class SignerWithProvider implements Signer {
...
     const intentMessage = messageWithIntent(
       IntentScope.TransactionData,
-      transactionBytes,
+      TransactionBlockBytes,
     );
     const signature = await this.signData(intentMessage);
 
     return {
-      transactionBytes: toB64(transactionBytes),
+      TransactionBlockBytes: toB64(TransactionBlockBytes),
```
Comment (Jordan-Mysten) [inline sdk/typescript/src/signers/signer-with-provider.ts:107]: Should be lowercase `T` here.

File: sdk/typescript/src/signers/types.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import { SerializedSignature } from '../cryptography/signature';
 
 export type SignedTransaction = {
-  transactionBytes: string;
+  TransactionBlockBytes: string;
```
Comment (Jordan-Mysten) [inline sdk/typescript/src/signers/types.ts:7]: Same here, lowercase the `t`

**Review by Jordan-Mysten — APPROVED** [review]
One nit for TS changes, but generally LGTM

File: sdk/typescript/src/signers/types.ts (hunk 2 above)
Comment (666lcz) [reply sdk/typescript/src/signers/types.ts:7]: Ah good catch!

**Pull request merged**